README.md

```diff
@@ -6,59 +6,67 @@ Isomorphic async tools for Preact.
 - Generate static HTML for your app using `prerender()`, waiting for `lazy()` components and data dependencies.
 - Implement async-aware client and server-side routing using `<Router>`, including seamless async transitions.
 
-### `lazy.js`
+## Routing
 
-Make a lazily-loaded version of a Component.
-`lazy()` takes an async function that resolves to a Component, and returns a wrapper version of that Component. The wrapper component can be rendered right away, even though the component is only loaded the first time it is rendered.
+`preact-iso` offers a simple router for Preact with conventional and hooks-based APIs. The `<Router>` component is async-aware: when transitioning from one route to another, if the incoming route suspends (throws a Promise), the outgoing route is preserved until the new one becomes ready.
 
 ```js
```

Comment:
I was thinking of this sending PR with this kitchen sink example:
```js
import { render } from 'preact';
import { LocationProvider, Router, ErrorBoundary, lazy, Route } from 'preact-iso';
import { useErrorBoundary } from 'preact/hooks';

// Asynchronous (returns a promise)
const Home = lazy(() => import('./routes/home.js'));
// Synchronous, is allowed as well (No lazy loading)
// import Home from './routes/home.js';
const Profile = lazy(() => import('./routes/profile.js'));
const PageNotFound = lazy(() => import('./routes/_404.js'));

function ErrorHandler(props) {
	// Documentation on useErrorBoundary() at https://preactjs.com/guide/v10/hooks/#useerrorboundary
  const [error] = useErrorBoundary(); 
  if (error) {
    return (
      <div>
        <h1>An unexpected error occurred:</h1>
        <p>{error.message}</p>
      </div>
    );
  }
	return <>{props.children}</>;
}

function App() {
	return (
		<LocationProvider>
			<ErrorBoundary>
				<ErrorHandler>
					<Router>
						<Home path="/" />
						{/* Alternative syntax - for TypeScript users */}
						<Route path="/profile" component={Profile} />
						{ /* `default` prop indicates a fallback route. Useful for 404 pages */ }
						<PageNotFound default />
					</Router>
				</ErrorHandler>
			</ErrorBoundary>
		</LocationProvider>
	);
);

render(<App />, document.body);
```

Reply:
Great idea!